crates/full-node/sov-db/src/storage_manager/nomt_based/groups.rs

```diff
@@ -61,25 +160,27 @@ where
         } = group;
 
         // Note: failure handling and data recovery will be implemented later.
-        self.state.commit(state)?;
-        self.accessory.write_schemas(&accessory)?;
+        self.merklized_state.commit(state)?;
+        self.accessory
+            .write_schemas(Arc::unwrap_or_clone(accessory))?;
         // Ledger goes last, as its data is used during the start.
         // So if ledger save failed, state and accessory will be synced from DA
-        self.ledger.write_schemas(&ledger)?;
+        self.ledger.write_schemas(Arc::unwrap_or_clone(ledger))?;
         // Historical data is committed the last, as in case of failure, it can be synced from the normal state,
```

Comment:
It looks like comments went ouf ot date.

Reply:
This wasn't changed in the PR, but now you've got me thinking - shouldn't the flat state be committed before the ledger (per this comment)?